anybox/recipe/odoo/base.py

```diff
@@ -279,14 +301,144 @@ def install_recipe_requirements(self):
                                           for dist in to_install]
-        sys.path.extend(self.recipe_requirements_paths)
+        # Some earlier processing leaves tmp dirs behind, that may
+        # mask what we just installed (especially harmful in case of pip)
+        sys.path[0:0] = self.recipe_requirements_paths
+
+    def merge_requirements(self, reqs=None):
+        """Merge eggs option with self.requirements.
+
+
+        TODO refactor all this: merge requirements is not idempotent, it
+        appends. Overall, this going back and forth between the serialized
+        form (self.options['eggs']) and the parsed version has growed too
+        much, up to the point where it's not natural at all.
+        """
+        if reqs is None:
+            reqs = self.requirements
+        serial = '\n'.join(reqs)
 
-    def merge_requirements(self):
-        """Merge eggs option with self.requirements."""
         if 'eggs' not in self.options:
-            self.options['eggs'] = '\n'.join(self.requirements)
+            self.options['eggs'] = serial
         else:
-            self.options['eggs'] += '\n' + '\n'.join(self.requirements)
+            self.options['eggs'] += '\n' + serial
+
+    def list_develops(self):
+        """At any point in time, list the projects that have been developed.
+
+        This can work as soon as the recipe is instantiated
+        (because the 'buildout' part) has already been executed.
+        In particular, it does not rely on the workingset init done by
+        :class:`zc.buildout.easy_install.Installer` and
+        can be used in precedence rules that need to be executed before calling
+        the Installer indirectly via ``zc.recipe.eggs``
+
+        :return: list of project names
+
+        Implementation simply lists the develop eggs directory
+        There's probably better to be done.
+        """
+        devdir_contents = (f.rsplit('.', 1) for f in os.listdir(
+            self.b_options['develop-eggs-directory']))
+        return [s[0] for s in devdir_contents if s[1] == 'egg-link']
+
+    def apply_odoo_requirements_file(self):
+        """Try and read Odoo's 'requirements.txt' and apply it.
+
+        This file appeared in the course of Odoo 8 lifetime. If not available,
+        a warning is issued, that's all.
+
+        Entries from the requirements file are applied if there is not already
+        an entry in the versions section for the same project.
+
+        A more interesting behaviour would be to apply then if they don't
+        contradict an existing entry in the versions section, but that's far
+        more complicated.
+        """
+        req_fname = 'requirements.txt'
+        req_path = join(self.openerp_dir, req_fname)
+        if not os.path.exists(req_path):
+            logger.warn("%r not found in this version of "
+                        "Odoo, although the configuration said to use it. "
+                        "Proceeding anyway.", req_fname)
+        else:
```

Comment:
If you want to save an indentation level for the rest of the func, returning after the warning would be fine as well.